file_writer_control/CommandChannel.py

```diff
@@ -0,0 +1,217 @@
+import threading
+from queue import Queue
+from file_writer_control.KafkaTopicUrl import KafkaTopicUrl
+from kafka import KafkaConsumer
+from kafka.errors import NoBrokersAvailable
+from typing import List, Union, Dict
+import atexit
+from datetime import datetime
+
+from file_writer_control.InThreadStatusTracker import InThreadStatusTracker
+from file_writer_control.WorkerStatus import WorkerStatus, WorkerState
+
+from file_writer_control.JobStatus import JobStatus, JobState
+from file_writer_control.CommandStatus import CommandStatus, CommandState
+from file_writer_control.InThreadStatusTracker import (
+    STATUS_MESSAGE_TIMEOUT,
+    COMMAND_STATUS_TIMEOUT,
+    JOB_STATUS_TIMEOUT,
+)
+
+
+def thread_function(host_port: str, topic: str, in_queue: Queue, out_queue: Queue):
+    """
+    Background thread for consuming Kafka messages.
+    :param host_port: The host + port of the Kafka broker that we are using.
+    :param topic: The Kafka topic that we are listening to.
+    :param in_queue: A queue for sending "exit" messages to the thread.
+    .. note:: The queue will exit upon the reception of the string "exit" on this queue.
+    :param out_queue: The queue to which status updates are published.
+    """
+    status_tracker = InThreadStatusTracker(out_queue)
+    while True:
+        try:
+            consumer = KafkaConsumer(
+                topic,
+                bootstrap_servers=host_port,
+                fetch_max_bytes=52428800 * 6,
+                consumer_timeout_ms=100,
+            )  # Roughly 300MB
+            break
+        except NoBrokersAvailable:
+            pass
```

Comment:
Even a simple error message for traceback is more useful than a pass. I may be missing something here, but I generally do not understand having pass in exceptions.

Reply:
In this particular case, we do not want to kill this background thread if the Kafka broker is not immediately available (due to some networking issue). I will clarify this a bit though.